google-cloud-spanner/src/main/java/com/google/cloud/spanner/SessionPool.java

```diff
@@ -288,6 +288,34 @@ public boolean next() throws SpannerException {
           }
         }
 
+        @Override
+        public boolean initiateStreaming(
+            AsyncResultSet.StreamMessageListener streamMessageListener) {
+          try {
+            boolean streamInitiated = super.initiateStreaming(streamMessageListener);
+            if (!streamInitiated) {
+              synchronized (lock) {
+                session.get().markUsed();
```

Comment:
I don't feel that this is correct. If `super.initiateStreaming()` returned false, then that means that no RPC was involved. That again means that the session was not used.

Reply:
yes. I removed this whole function. it's not relevant for initiate streaming.